src/Data/Json/Extended/Signature/Core.purs

```diff
@@ -0,0 +1,153 @@
+module Data.Json.Extended.Signature.Core
+  ( EJsonF(..)
+  ) where
+
+import Prelude
+
+import Data.Eq1 (class Eq1)
+import Data.Ord1 (class Ord1)
+
+import Data.Foldable as F
+import Data.HugeNum as HN
+import Data.Int as Int
+import Data.List as L
+import Data.Map as Map
+import Data.Tuple as T
+
+-- | The signature endofunctor for the EJson theory.
+data EJsonF a
+  = Null
+  | String String
+  | Boolean Boolean
+  | Integer Int
+  | Decimal HN.HugeNum
+  | String String
+  | Timestamp String
+  | Date String
+  | Time String
+  | Interval String
+  | ObjectId String
+  | Array (Array a)
+  | OrderedSet (Array a)
+  | Object (Array (T.Tuple a a))
+
+instance functorEJsonF ∷ Functor EJsonF where
+  map f x =
+    case x of
+      Null → Null
+      String str → String str
+      Boolean b → Boolean b
+      Integer i → Integer i
+      Decimal a → Decimal a
+      Timestamp ts → Timestamp ts
+      Date d → Date d
+      Time t → Time t
+      Interval i → Interval i
+      ObjectId oid → ObjectId oid
+      Array xs → Array $ f <$> xs
+      OrderedSet xs → OrderedSet $ f <$> xs
+      Object xs → Object $ bimapTuple f <$> xs
+    where
+      bimapTuple f (T.Tuple a b) =
```

Comment:
I figured I might as well not incur an entire dependency to get one function that's easily implemented, but I just checked and we already transitively depend on this anyway. So I'll switch to `bimap`.